src/Mvc/Mvc.Core/src/MvcCoreLoggerExtensions.cs

```diff
@@ -909,6 +909,13 @@ public static void NoActionsMatched(this ILogger logger, IDictionary<string, obj
         }
     }
 
+    [LoggerMessage(
```

Comment:
Let's move this to DefaultActionDescriptorCollectionProvider.cs. The pattern is usually to put logs under a nested static class named Log e.g.

```C#
public partial class DefaultActionDescriptorCollectionProvider
{
    public void SomeAPI()
    {
        Log.MyLog(_logger, ...);
    }
    
    public static partial class Log
    {
        [LoggerMessage(...)]
        public static partial void MyLog(ILogger logger, ...);
    }
}
```

There's a [PR that updates MVC](https://github.com/dotnet/aspnetcore/pull/38658) to use this pattern that I really need to review soon